grails-app/services/org/pih/warehouse/core/UserService.groovy

```diff
@@ -149,10 +149,21 @@ class UserService {
         return false
     }
 
+    Boolean hasHighestRoleAuthenticated(User u, String locationId) {
```

Comment:
I know this should be obvious, but I'm struggling with the naming of this method. Is this saying, the user at the given should at least by an authenticated user?

Also, this should take the roleType as an argument so it can used in other places.
```
hasHighestRole(User user, String locationId, RoleType )
```

Reply:
Ok, I'll modify this one. It is saying that the user is not in a higher role than ... It is done like this to show simpler menu for user and to block going to dashboard from clicking logo etc.